plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -251,14 +257,18 @@ private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
         val (name, version) = parseNameAndVersion(entry)
         versionForName[name] = checkNotNull(version)
 
-        val dependencies = node[entry]?.map { it.textValue().substringBefore(" ") }.orEmpty()
+        val dependencies = node[entry]?.map { depNode ->
+            val (depName, depVersion) = parseNameAndVersion(depNode.textValue())
+            depName.also { if (depVersion != null) versionForName[it] = depVersion }
```

Comment:
Can line 262 be simplified to just `if (depVersion != null) versionForName[it] = depVersion` ?

Reply:
Or maybe `depVersion?.let { versionForName[depName] = it }`

Reply:
No, as `depName` needs to be returned for `map`.